web/src/user_status.ts

```diff
@@ -43,21 +44,44 @@ const user_status_param_schema = z.record(z.string(), user_status_schema);
 const user_info = new Map<number, string>();
 const user_status_emoji_info = new Map<number, UserStatusEmojiInfo>();
 
+let user_status_end_time: number;
+
+export function set_user_status_end_time(status_end_time: string): void {
+    user_status_end_time = Number.parseInt(status_end_time, 10);
+}
+
+export function get_user_status_end_time(): number {
+    return user_status_end_time;
+}
+
 export function server_update_status(opts: {
     status_text: string;
     emoji_name: string;
     emoji_code: string;
     reaction_type?: string;
+    status_end_time?: number;
     success?: () => void;
 }): void {
-    void channel.post({
-        url: "/json/users/me/status",
-        data: {
+    let data;
+    if (opts.status_end_time) {
+        data = {
             status_text: opts.status_text,
             emoji_name: opts.emoji_name,
             emoji_code: opts.emoji_code,
             reaction_type: opts.reaction_type,
-        },
+            status_end_time: opts.status_end_time,
+        };
+    } else {
+        data = {
+            status_text: opts.status_text,
+            emoji_name: opts.emoji_name,
+            emoji_code: opts.emoji_code,
+            reaction_type: opts.reaction_type,
+        };
+    }
```

Comment:
Hmm, it seems weird to need this if/else clause that some of these other parameters are variable.

Reply:
yeah, you're right. updated it.